pkg/artifacts/registry.go

```diff
@@ -23,21 +23,29 @@ func NewRegistryPuller() *RegistryPuller {
 }
 
 func (p *RegistryPuller) Pull(ctx context.Context, ref string) ([]byte, error) {
-	registry, err := content.NewRegistry(content.RegistryOptions{})
+	var art registry.Artifact
+	err := art.SetURI(ref)
 	if err != nil {
-		return nil, fmt.Errorf("creating registry: %w", err)
+		return nil, err
 	}
-	store := content.NewMemory()
 
-	_, err = oras.Copy(ctx, registry, ref, store, "")
+	certificates, err := registry.GetCertificates("registry.crt")
 	if err != nil {
-		return nil, fmt.Errorf("pulling artifact %q: %s", ref, err)
+		return nil, err
 	}
```

Comment:
What if no cert is required like pulling from private ecr or insecure local registry? I think we need a way to know if cert is intended to be used and errors out accordingly.

Reply:
This will return an empty X509 struct if the file is empty which will be the case in the pod if the volume secret is mounted. I'm not a fan of completely ignoring the error though.

Reply:
okay, I passed a logger down, so it can just log the error and ignore it. I also updated the name of the file to match the helm chart